lib/rack/request.rb

```diff
@@ -497,7 +497,8 @@ def GET
       # multipart/form-data.
       def POST
```

Comment:
This method seems a little messy, I might try to tidy it up, if we approve of this general (PR) direction.

Reply:
Okay, I tidied this up.